<launch>
  <!-- Specify Occ. Grid args. -->
  <arg name="width" value="100"/>
  <arg name="height" value="100"/>
  <arg name="resolution" value="0.25"/>
  <arg name="percent_occupied" value="0.08"/>
  <!-- Specify RRT args. -->
  <arg name="x_initial" value="0.3"/>
  <arg name="y_initial" value="0.3"/>
  <arg name="x_goal" value="10."/>
  <arg name="y_goal" value="18."/>
  <arg name="goal_tol" value="0.25"/>
  <arg name="max_rrt_iter" value="5000"/>
  <arg name="rrt_ext_distance" value="0.5"/>

  <!-- Launch Occupancy Grid Generator Node -->
  <node type="occ_grid_generator_node" name="occ_grid_generator_node" pkg="rrt-global-planner">
    <param name="width" value="$(arg width)"/>
    <param name="height" value="$(arg height)"/>
    <param name="resolution" value="$(arg resolution)"/>
    <param name="percent_occupied" value="$(arg percent_occupied)"/>
  </node>
  
  <!-- Launch RRT Publisher Node -->
  <node type="rrt_node" name="rrt_node" pkg="rrt-global-planner" output="screen">
    <param name="x_initial" type="double" value="$(arg x_initial)"/>
    <param name="y_initial" type="double" value="$(arg y_initial)"/>
    <param name="x_goal" type="double" value="$(arg x_goal)"/>
    <param name="y_goal" type="double" value="$(arg y_goal)"/>

    <param name="goal_tol" type="double" value="$(arg goal_tol)"/>
    <param name="K_in" type="int" value="$(arg max_rrt_iter)"/>
    <param name="d" type="double" value="$(arg rrt_ext_distance)"/>
  </node>

  <!-- Launch RViz -->
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find rrt-global-planner)/config/rrt.rviz" />
</launch>